File: libs/text_utils.py
def remove_newlines_and_spacing(text):
    """
    Remove newlines and extra spaces from the given text.

    Args:
        text (str): The input text.

    Returns:
        str: The text without newlines and extra spaces.
    """
    return text.replace("\n", "").strip()

def standardize_text(text):
    """
    Standardize the given text by removing newlines, extra spaces, and normalizing spacing.

    Args:
        text (str): The input text.

    Returns:
        str: The standardized text.
    """
    text = remove_newlines_and_spacing(text)
    text = " ".join(text.split())
    return text

from datetime import datetime, timezone

def unixtimestamp_to_date(timestamp):
    """
    Convert a Unix timestamp to a date string in the format 'YYYY-MM-DD'.

    Args:
        timestamp (int): The Unix timestamp.

    Returns:
        str: The date string in the format 'YYYY-MM-DD'.
    """
    ts = int(timestamp)
    return datetime.fromtimestamp(ts, timezone.utc).strftime('%Y-%m-%d')

File: vue-module/src/utils/textUtils.ts
/**
 * Remove newlines and extra spacing from text
 */
export function removeNewlinesAndSpacing(text: string): string {
  return text.replace(/\n/g, '').trim();
}

/**
 * Standardize text by removing newlines and normalizing whitespace
 */
export function standardizeText(text: string): string {
  const cleaned = removeNewlinesAndSpacing(text);
  return cleaned.replace(/\s+/g, ' ');
}

/**
 * Convert Unix timestamp to YYYY-MM-DD date string
 */
export function unixTimestampToDate(timestamp: string | number): string {
  const ts = typeof timestamp === 'string' ? parseInt(timestamp) : timestamp;
  return new Date(ts * 1000).toISOString().split('T')[0];
}

/**
 * Get date template for the last 366 days
 */
export function getDateTemplate(): Record<string, number> {
  const today = new Date();
  const last366Days = new Date(today.getTime() - 366 * 24 * 60 * 60 * 1000);
  
  const data: Record<string, number> = {};
  data[today.toISOString().split('T')[0]] = 0;
  data[last366Days.toISOString().split('T')[0]] = 0;
  
  return data;
}

/**
 * Format date as YYYY-MM-DD
 */
export function formatDate(date: Date): string {
  return date.toISOString().split('T')[0];
}